app/src/main/java/com/google/jetpackcamera/MainActivity.kt

```diff
@@ -101,13 +107,67 @@ class MainActivity : ComponentActivity() {
                             modifier = Modifier.fillMaxSize(),
                             color = MaterialTheme.colorScheme.background
                         ) {
-                            JcaApp(onPreviewViewModel = { previewViewModel = it })
+                            JcaApp(
+                                onPreviewViewModel = { previewViewModel = it },
+                                contentResolver = contentResolver,
+                                imageCaptureUri = uri,
+                                onImageCapture = { event ->
+                                    val shouldFinishWithResult = isExternalImageCapture()
+                                    when (event) {
+                                        is CameraUseCase.ImageCaptureEvent.ImageSaved -> {
+                                            if (shouldFinishWithResult) {
+                                                setResult(RESULT_OK)
+                                                finish()
+                                            }
+                                        }
+
+                                        is CameraUseCase.ImageCaptureEvent.ImageCaptureError -> {
+                                            if (shouldFinishWithResult) {
+                                                setResult(RESULT_CANCELED)
+                                                finish()
+                                            }
+                                        }
+                                    }
+                                }
+                            )
                         }
                     }
                 }
             }
         }
     }
+
+    private fun getImageCaptureUriIfExists(): Uri? {
+        val isExternalImageCapture = isExternalImageCapture()
+        val uri = if (!isExternalImageCapture || intent.extras == null ||
+            !intent.extras!!.containsKey(MediaStore.EXTRA_OUTPUT)
+        ) {
+            null
+        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            intent.extras!!.getParcelable(
+                MediaStore.EXTRA_OUTPUT,
+                Uri::class.java
+            )
+        } else {
+            @Suppress("DEPRECATION")
+            intent.extras!!.getParcelable(MediaStore.EXTRA_OUTPUT)
```

Comment:
It looks like you should be using `getParcelable(EXTRA_OUTPUT, Uri::class.java)` here on API 33+.

Reply:
this is for version below 33 where that api was unavailable. Our min version is 21